src/gui/DatabaseWidget.cpp

```diff
@@ -1548,7 +1550,11 @@ bool DatabaseWidget::saveAs()
             // Ensure we don't recurse back into this function
             m_db->setReadOnly(false);
             m_db->setFilePath(newFilePath);
-
+#ifdef WITH_XC_KEESHARE
+            // We need to reregister the database to allow exports
+            // from a newly created database
+            KeeShare::instance()->connectDatabase(m_db, m_db);
```

Comment:
I don't necessarily like passing the "old" database pointer into connectDatabase. The keeshare observer should listen for the databases' ```QObject::destroyed``` signal and handle any necessary cleanup. It is unnecessary to disconnect signal/slots from an object being deleted.